src/main/java/mate/academy/spring/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.ShoppingCartDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ShoppingCartDaoImpl extends AbstractDao<ShoppingCart> implements ShoppingCartDao {
+    public ShoppingCartDaoImpl(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<ShoppingCart> query = session.createQuery("FROM ShoppingCart WHERE user = "
+                    + ":user", ShoppingCart.class);
+            query.setParameter("user", user);
+            return query.uniqueResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Cannot find shopping cart using user ", e);
+        }
+    }
+
+    @Override
+    public void update(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
```

Comment:
The transaction could be not rollbacked if an exception occurs because the session will be closed. So let's not use try with resources